pydra/engine/tests/test_shelltask_inputspec.py

```diff
@@ -537,6 +537,217 @@ def test_shell_cmd_inputs_mandatory_1():
+                        "argstr": "",
+                        "mandatory": True,
+                    },
+                ),
+            ),
+            (
+                "outA",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "position": 2,
+                        "help_string": "outA",
+                        "argstr": "-o",
+                        "output_file_template": "{inpA}_out",
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="executable", input_spec=my_input_spec, inpA="inpA"
+    )
+    # outA has position, so is a part of the command line
+    assert shelly.cmdline == "executable inpA -o inpA_out"
+
+
+def test_shell_cmd_inputs_template_1a():
+    """ additional inputs, one uses output_file_template (without position)"""
+    my_input_spec = SpecInfo(
+        name="Input",
+        fields=[
+            (
+                "inpA",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "position": 1,
+                        "help_string": "inpA",
+                        "argstr": "",
+                        "mandatory": True,
+                    },
+                ),
+            ),
+            (
+                "outA",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "help_string": "outA",
+                        "argstr": "-o",
+                        "output_file_template": "{inpA}_out",
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellSpec,),
+    )
+
+    shelly = ShellCommandTask(
+        executable="executable", input_spec=my_input_spec, inpA="inpA"
+    )
+    # outA has no position, so is not a part of the command line
+    assert shelly.cmdline == "executable inpA"
+
+
+def test_shell_cmd_inputs_template_2():
+    """ additional inputs, one uses output_file_template (and position)"""
+    my_input_spec = SpecInfo(
+        name="Input",
+        fields=[
+            (
+                "inpB",
+                attr.ib(
+                    type=str,
+                    metadata={"position": 1, "help_string": "inpB", "argstr": ""},
+                ),
+            ),
+            (
+                "outB",
+                attr.ib(
+                    type=str,
+                    metadata={
+                        "position": 2,
+                        "help_string": "outB",
+                        "argstr": "-o",
+                        "output_file_template": "{inpB}_out",
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellSpec,),
+    )
+
+    shelly = ShellCommandTask(executable="executable", input_spec=my_input_spec)
+    # outA has position, so is a part of the command line
```

Comment:
this comment is a bit weird for this use case, since it actually doesn't output anything given lack of input.